lib/prometheus/business_collector.rb

```diff
@@ -84,6 +101,37 @@ def collect
     @client_policies.observe(
       Policy.where(account_id: client_accounts.select(:id)).count
     )
+
+    Policy.joins(:profiles)
+          .select(:account_id, "profiles.ref_id").distinct
+          .group(:ref_id)
+          .count(:account_id).each do |ref_id, count|
+      @total_policies_by_account.observe(count, ref_id: ref_id)
+    end
+
+    Policy.where(account_id: client_accounts.pluck(:id))
+          .joins(:profiles)
+          .select(:account_id, "profiles.ref_id").distinct
+          .group(:ref_id)
+          .count(:account_id).each do |ref_id, count|
+      @client_policies_by_account.observe(count, ref_id: ref_id)
+    end
+
+    Policy.joins(:policy_hosts).joins(:profiles)
```

Comment:
Just a note, that policy hosts and profiles do not have a direct relationship together. It would join hosts with any of the policy profiles. This works thanks to the distinct counting and the fact that a policy uses exactly one profile ref_id.

Reply:
I am aware, that's why I was using the subquery-querying :wink: that looked more trivial, but you asked for this :see_no_evil: but I like this more now :laughing: